workers/unity/Assets/Gdk/Core/Error/ForwardingDispatcher.cs

```diff
@@ -0,0 +1,67 @@
+using Improbable.Worker;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core
+{
+    public class ForwardingDispatcher : ILogDispatcher
+    {
+        private Connection connection;
+
+#if UNITY_EDITOR
+        private readonly LoggingDispatcher loggingDispatcher = new LoggingDispatcher();
+#endif
+
+        public void SetConnection(Connection connection)
+        {
+            this.connection = connection;
+        }
+
+        public void HandleLog(LogType type, LogEvent logEvent)
+        {
+#if UNITY_EDITOR
+            if (Application.isEditor)
+            {
+                loggingDispatcher.HandleLog(type, logEvent);
+            }
+#endif
+
+            if (connection == null)
+            {
+                return;
+            }
+
+            LogLevel logLevel;
+            var message = logEvent.ToString();
+
+            switch (type)
+            {
+                case LogType.Error:
+                    logLevel = LogLevel.Error;
+                    break;
+                case LogType.Warning:
+                    logLevel = LogLevel.Warn;
+                    break;
+                case LogType.Log:
+                    logLevel = LogLevel.Info;
+                    break;
+                case LogType.Assert:
+                    logLevel = LogLevel.Error;
+                    break;
+                case LogType.Exception:
+                    logLevel = LogLevel.Fatal;
+                    break;
+                default:
+                    logLevel = LogLevel.Debug;
+                    break;
+            }
+
+            Collections.Option<EntityId> entityId;
+            if (logEvent.Data.ContainsKey(IdType.EntityId.ToString()))
+            {
+                entityId = new EntityId((long) logEvent.Data[IdType.EntityId.ToString()]);
```

Comment:
What happens here if someone accidentally stores a non-`long` type, but uses the IdType.EntityID? This will throw and blow up!

Reply:
Is it worth making a helper function for getting well known bits of metadata?